src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -23,11 +24,25 @@ def _osrandom_rand_status():
     return 1
 
 
+class _ConditionalLibrary(object):
+    def __init__(self, lib, conditional_names):
+        excluded_names = set()
+        for condition, names in conditional_names.items():
+            if not getattr(lib, condition):
+                excluded_names |= set(names)
+
+        for attr in dir(lib):
+            if attr in excluded_names:
+                continue
+            else:
```

Comment:
This is simpler if you flip teh condition around `if attr not in excluded_names: setattr..`